src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1981,16 +1981,24 @@ export default class AircraftModel {
             holdParameters.inboundHeading = bearingToHoldFix;
         }
 
-        const { inboundHeading } = holdParameters;
+        const { inboundHeading, legLength } = holdParameters;
         const outboundHeading = radians_normalize(inboundHeading + Math.PI);
         const groundTrack = radians_normalize(this.groundTrack);
         const offset = getOffset(this, waypointRelativePosition, inboundHeading);
-        const holdLegDurationInMinutes = holdParameters.legLength.replace('min', '');
-        const holdLegDurationInSeconds = holdLegDurationInMinutes * TIME.ONE_MINUTE_IN_SECONDS;
         const gameTime = TimeKeeper.accumulatedDeltaTime;
         const isPastFix = offset[1] < 1 && offset[2] < 2;
         const isTimerSet = holdParameters.timer !== INVALID_NUMBER;
         const isTimerExpired = isTimerSet && gameTime > holdParameters.timer;
+        let holdLegDurationInSeconds;
+
+        if (legLength.indexOf('min') !== -1) {
+            const holdLegDurationInMinutes = legLength.replace('min', '');
+            holdLegDurationInSeconds = holdLegDurationInMinutes * TIME.ONE_MINUTE_IN_SECONDS;
+        } else {
+            // Leg is a distance, use the ground speed to determine the duration
+            const holdLegDistance = legLength.replace('nm', '');
+            holdLegDurationInSeconds = (holdLegDistance / this.groundSpeed) * TIME.ONE_HOUR_IN_SECONDS;
```

Comment:
What if some bozo enters `8000m` for the leg length? Do we validate that somewhere to be either `min` or `nm`?

Reply:
We want the groundspeed when the aircraft is established on the outbound leg, or else the distance will be wrong. Is this when this is calculated?

Reply:
> What if some bozo enters `8000m` for the leg length? Do we validate that somewhere to be either `min` or `nm`?

Good shout, but yes, that's validated for in the `argumentParser.isLegLengthArg` function

> We want the groundspeed when the aircraft is established on the outbound leg, or else the distance will be wrong. Is this when this is calculated?

Good question. `_calculateTargetedHeadingHold` appears to be called every tick. To verify this I watched the value for `this.groundSpeed`, the `holdLegDurationInSeconds` is calculated for an up-to-date GS value. So even if I order the hold well before the fix is reached, it will use the GS when the a/c is `MAXIMUM_ANGLE_CONSIDERED_ESTABLISHED_ON_HOLD_COURSE`.

For example if the aircraft is long before the fix (cruising at 250KIAS) and I order 165KIAS (about 180KTAS) and a leg length of 3NM (aiming for 60s leg length). The `holdLegDurationInSeconds` is calculated to be about 62s.

In fact, to make thiings more efficient, `holdLegDurationInSeconds` only needs to be calculated if `currentWaypoint.setHoldTimer` needs to be called (as that's the only place it's used).

``` js
if (abs(groundTrack - outboundHeading) < PERFORMANCE.MAXIMUM_ANGLE_CONSIDERED_ESTABLISHED_ON_HOLD_COURSE && !isTimerSet) {
    let holdLegDurationInSeconds;

    if (legLength.indexOf('min') !== -1) {
        const holdLegDurationInMinutes = legLength.replace('min', '');
        holdLegDurationInSeconds = holdLegDurationInMinutes * TIME.ONE_MINUTE_IN_SECONDS;
    } else {
        // Leg is a distance, use the ground speed to determine the duration
        const holdLegDistance = legLength.replace('nm', '');
        holdLegDurationInSeconds = (holdLegDistance / this.groundSpeed) * TIME.ONE_HOUR_IN_SECONDS;
    }

    console.log(isPastFix, this._isEstablishedOnHoldingPattern, this.groundSpeed, holdLegDurationInSeconds);
    currentWaypoint.setHoldTimer(gameTime + holdLegDurationInSeconds);
}
```

Reply:
Both holding at BPK @ 165 KIAS. One with a leg length of 3NM, the other with a leg length of 1min
![image](https://user-images.githubusercontent.com/3430117/66136990-48f04e80-e5f4-11e9-81a3-2f1a3f9489a4.png)